src/components/BoardView.js

```diff
@@ -17,16 +17,38 @@ const BACKGROUNDS_NR = 5;
 
 const randomBg = backgrounds[random(BACKGROUNDS_NR)];
 
+const randomCardIds = (cardCount, imgArr) => {
+  let array = shuffle(imgArr).splice(0, cardCount / 2);
+
+  const doubledArray = array.concat(array);
+
+  return shuffle(doubledArray);
+};
+
 class BoardView extends Component {
   state = {
-    cardCount: range(CARD_COUNT)
+    cardCount: []
   };
 
+  componentDidUpdate(oldProps) {
+    if (this.props !== oldProps ) {
+      this.setState({
+        cardCount: randomCardIds(CARD_COUNT, this.props.photoArray)
```

Comment:
Why would you randomize something, that by name is an integer?

Reply:
I do not know what you mean. Do you suggest changing the name of `randomCardIds`?

Reply:
you have "cardCount" a value suggesting to be an INTEGER that is set from a function suggesting to produce a COLLECTION.